src/platform/nrfconnect/ConnectivityManagerImpl.h

```diff
@@ -81,6 +85,44 @@ class ConnectivityManagerImpl final : public ConnectivityManager,
     // the implementation methods provided by this class.
     friend class ConnectivityManager;
 
+public:
+    // Generic network status checkers
+    bool IsIPv6NetworkEnabled()
+    {
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
+        return IsThreadEnabled();
+#elif CHIP_DEVICE_CONFIG_ENABLE_WIFI
+        return IsWiFiStationEnabled();
+#endif
```

Comment:
For thread + Wi-Fi configuration, these checks should be ORed, right?

```suggestion
        return false
#if CHIP_DEVICE_CONFIG_ENABLE_THREAD
            || IsThreadEnabled();
#endif
#if CHIP_DEVICE_CONFIG_ENABLE_WIFI
            || IsWiFiStationEnabled();
#endif
            ;
```

Reply:
Ok, that's right. But this applies to the `IsEnable()` only, right? When it comes to `IsProvisioned()` we cannot have both Thread and Wi-Fi provisioned at the same time.

Reply:
Actually, we can, but even if we couldn't, you would need to support the case where we have both Thread + Wi-Fi support **enabled**, so it's still worth ORing the return values in such a case.

Reply:
Yes, definitely we should do OR in the case of IsEnable(). The only thing I have doubts about is the possibility of simultaneous provisioning of both backends. Is it really a case for our samples? It sounds more like a TBR case to me.

Reply:
For our samples, no, but it is possible to have two NetworkCommissioning instances for Thread + WiFi and have both provisioned. What would you use instead of OR even if this wasn't possible?

Reply:
Discussed f2f.